zppy/global_time_series.py

```diff
@@ -52,6 +53,48 @@ def global_time_series(config, scriptDir, existing_bundles, job_ids_file):
             if skip:
                 continue
 
+            # Handle legacy parameter
+            if (not c["plots_original"]) and c["plot_names"]:
+                c["plots_original"] = c["plot_names"]
+
+            # Determine which components are needed
+            c["use_atm"] = False
+            c["use_lnd"] = False
+            c["use_ocn"] = False
+            c["use_sea_ice"] = False
+            if c["plots_original"]:
+                c["use_atm"] = True
+                if not c["atmosphere_only"]:
```

Comment:
`atmosphere_only` originally meant "not the ocean plots" but that is less straightforward now with the other components being added in. What I did was  make it so "atmosphere_only" applies _strictly_ to the original plots and behaves exactly as before -- that is, skip the ocean plots in the _original_ plot list only.

Question 3) Is that a reasonable choice? Or do we want to handle `atmosphere_only` differently? (We could also deprecate it, but it needs to remain for backwards compatibility).

Reply:
what options do we have currently? I think making the plotted variables match this option makes better sense to me, such as land_only, atm_only, ocean_only, land_atm (F compset), all (B compset), etc.

Reply:
@BunnyVon I think the best choice is to simply deprecate `atmosphere_only`. 

Option 1: deprecate `atmosphere_only`
---
With this pull request, these are the relevant parameters that can be set in the cfg. Excerpted from the `default.ini` file.
```
[global_time_series]
# Set to True to skip figures requiring ocean data; only affects original plots
atmosphere_only = boolean(default=False)
# Legacy name for plots_original; kept for backwards compatibility
plot_names = string(default="net_toa_flux_restom,global_surface_air_temperature,toa_radiation,net_atm_energy_imbalance,change_ohc,max_moc,change_sea_level,net_atm_water_imbalance")
# The names of the plots you want displayed
plots_original = string(default="net_toa_flux_restom,global_surface_air_temperature,toa_radiation,net_atm_energy_imbalance,change_ohc,max_moc,change_sea_level,net_atm_water_imbalance")
# The names of the extra plots you want displayed (i.e., define specific variables)
plots_atm = string(default="")
plots_lnd = string(default="")
plots_ocn = string(default="")
plots_sea_ice = string(default="")
```

`plot_names` and `atmosphere_only` would need to be kept for backwards compatibility, but I think we can deprecate them. `plot_names` behaves exactly as `plots_original`. And `atmosphere_only`, if used, will behave as it has in the past, with no new functionality. That is, it would only affect the original 8 plots.

Option 2: `<component>-only` options for all components
---
This would essentially be equivalent to setting: 
```
plots_<component>=[var1, var2,...]
plots_<other1>=[]
plots_<other2>=[]
...
```
However, there is the notable exception that `plots_original` has 5 atm plots and 3 ocn plots. So, `atmosphere_only` essentially turns off the 3 ocn plots there and a new `ocean_only` parameter would turn off the 5 atm plots there, without users needing to modify the variable list itself.

I can see it providing limited use, but it's basically adding 3 new parameters (`land_only`, `ocean_only`, `sea_ice_only`) for not much gain. It seems simpler for users to just comment out components they don't want plotted.